### PR TITLE
Adjust AD integration to represent new reality

### PR DESCRIPTION
**Pull request by ngaranko**
Looks like new Azure AD integration contains different set of parameters than before.
This PR adjusts our code to work with AzureAD as Identity Provider.

Also fixing minor issue with expired token errors.

### CODE REVIEW

File: authorization_django/middleware.py [hunk 1]
```diff
@@ -166,11 +166,14 @@ def get_claims(jwt):
                 'scopes': {convert_scope(r) for r in claims['realm_access']['roles']},
                 'claims': claims
             }
-        elif claims.get('scp') and claims.get('preferred_username'):
-            # Microsoft token structure
+        elif claims.get('groups') and (claims.get('unique_name') or claims.get('un')):
```
Comment (drakonen) [inline authorization_django/middleware.py:169]: Ik zou hier extra haakjes omheen zetten om duidelijker te maken hoe de condities gegropeerd zijn.

Comment (ngaranko) [reply authorization_django/middleware.py:169]: Daar staan ze, toch? Of bedoel je iets anders?

Comment (drakonen) [reply authorization_django/middleware.py:169]: :man_facepalming:

File: authorization_django/middleware.py [hunk 2]
```diff
@@ -166,11 +166,11 @@ def get_claims(jwt):
...
```
Comment (vdboor) [inline authorization_django/middleware.py:169]: Why not support both? Or was the previous already designed for Azure?

Comment (ngaranko) [reply authorization_django/middleware.py:169]: Previous was designed for azure indeed, but never used (surprise!) and apparently azure is not using it anymore :/

**Pull request merged**